src/model/writer.js

```diff
@@ -789,101 +860,71 @@ export default class Writer {
-	 * 		setMarker( markerName, range, { usingOperation: true } );
-	 *
-	 * Create marker with a unique id using operation:
-	 *
-	 * 		setMarker( range, { usingOperation: true } );
-	 *
-	 * Create marker directly without using operations:
-	 *
-	 * 		setMarker( range )
+	 * 		setMarker( marker, { range, usingOperation: true } );
+	 * 		setMarker( markerName, { range, usingOperation: true } );
 	 *
 	 * Change marker's option (start using operations to manage it):
 	 *
-	 * 		setMarker( marker, { usingOperation: true } );
-	 *
-	 * Note: For efficiency reasons, it's best to create and keep as little markers as possible.
+	 * 		updateMarker( marker, { usingOperation: true } );
 	 *
 	 * @see module:engine/model/markercollection~Marker
-	 * @param {module:engine/model/markercollection~Marker|String} [markerOrName]
-	 * Name of a marker to create or update, or `Marker` instance to update, or range for the marker with a unique name.
-	 * @param {module:engine/model/range~Range} [range] Marker range.
-	 * @param {Object} [options]
-	 * @param {Boolean} [options.usingOperation=false] Flag indicated whether the marker should be added by MarkerOperation.
+	 * @param {String} markerOrName Name of a marker to update, or a marker instance.
+	 * @param {Object} options
+	 * @param {module:engine/model/range~Range} [options.range] Marker range to update.
+	 * @param {Boolean} [options.usingOperation] Flag indicated whether the marker should be added by MarkerOperation.
 	 * See {@link module:engine/model/markercollection~Marker#managedUsingOperations}.
 	 * @returns {module:engine/model/markercollection~Marker} Marker that was set.
 	 */
-	setMarker( markerOrNameOrRange, rangeOrOptions, options ) {
+	updateMarker( markerOrName, options ) {
 		this._assertWriterUsedCorrectly();
 
-		let markerName, newRange, usingOperation;
+		const markerName = typeof markerOrName == 'string' ? markerOrName : markerOrName.name;
 
-		if ( markerOrNameOrRange instanceof Range ) {
-			markerName = uid();
-			newRange = markerOrNameOrRange;
-			usingOperation = !!rangeOrOptions && !!rangeOrOptions.usingOperation;
-		} else {
-			markerName = typeof markerOrNameOrRange === 'string' ? markerOrNameOrRange : markerOrNameOrRange.name;
+		const currentMarker = this.model.markers.get( markerName );
 
-			if ( rangeOrOptions instanceof Range ) {
-				newRange = rangeOrOptions;
-				usingOperation = !!options && !!options.usingOperation;
-			} else {
-				newRange = null;
-				usingOperation = !!rangeOrOptions && !!rangeOrOptions.usingOperation;
-			}
+		if ( !currentMarker ) {
+			/**
+			 * Marker with provided name does not exists.
+			 *
+			 * @error writer-updateMarker-marker-not-exists
+			 */
+			throw new CKEditorError( 'writer-updateMarker-marker-not-exists: Marker with provided name does not exists.' );
 		}
 
-		const currentMarker = this.model.markers.get( markerName );
+		const range = options && options.range;
+		const usingOperation = !!options && !!options.usingOperation;
```

Comment:
Maybe I do not understand something or was not clear enough, but if I call `writer.updateMarker( name, { range } );` (with `usingOperation` not specified), then `usingOperation` should **not change**. So only range should change. This code looks like it changes `usingOperation` to `false` in such case. Also, I can not find a test which checks this.